diff --git a/src/common/pico_base/include/pico/assert.h b/src/common/pico_base/include/pico/assert.h
index 8910ebd..229f2cc 100644
--- a/src/common/pico_base/include/pico/assert.h
+++ b/src/common/pico_base/include/pico/assert.h
@@ -15,7 +15,9 @@
 
 extern "C" {
 #else
-#include <assert.h>
+//#include <assert.h>
+#define static_assert(...)
+#define assert(...)
 #endif
 
 // PICO_CONFIG: PARAM_ASSERTIONS_ENABLE_ALL, Global assert enable, type=bool, default=0, group=pico_base
